test/e2e/reporters/progress.go

```diff
@@ -0,0 +1,148 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package reporters
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/onsi/ginkgo/config"
+	"github.com/onsi/ginkgo/types"
+	"k8s.io/klog"
+)
+
+// ProgressReporter is a ginkgo reporter which tracks the total number of tests to be run/passed/failed/skipped.
+// As new tests are completed it updates the values and prints them to stdout and optionally, sends the updates
+// to the configured URL.
+type ProgressReporter struct {
+	LastMsg string `json:"msg"`
+
+	NumToBeRun   int `json:"total"`
+	NumCompleted int `json:"completed"`
+	NumSkipped   int `json:"skipped"`
+	NumFailed    int `json:"failed"`
+
+	Failures []string `json:"failures,omitempty"`
+
+	progressURL string
+	client      *http.Client
+}
+
+// NewProgressReporter returns a progress reporter which posts updates to the given URL.
+func NewProgressReporter(progressUpdateURL string) *ProgressReporter {
+	c := &http.Client{
+		Timeout: time.Second * 10,
```

Comment:
It looks great! But are you thinking about making `Timeout ` configurable?

Reply:
I don't really anticipate something like that being particularly configurable. It is really just specifying a timeout so it doesn't hang forever. And regardless, the penalty for one slow request is that it gets dropped effectively which is OK IMO for one or two events.

Do you see a reason it should be a higher value? I wouldn't care about increasing it to a high, but reasonable, value.